controllers/__init__.py

```diff
@@ -47,14 +48,17 @@ def bootstrap_indexes(self) -> None:
                     f"Created DB index {keys}, collection: {collection}"
                 )
 
-    def get_circuits(self, archived: Optional[bool] = False) -> Dict:
+    def get_circuits(self, archived: Optional[bool] = False,
+                     match: dict = None) -> Dict:
```

Comment:
Let's keep this more specific, you could take `metadata_pairs: Optional[list[tuple[str, Any]]] = None` and then build the `match_filters["$match"][key] = value` accordingly when it's not `None`

Reply:
Schema has been changed